scripts/check_register.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+import csv
+from collections import defaultdict
+import operator
+import argparse
+import sys
+import os
+import get_ps_list
+
+def parse_args(args_):
+    parser = argparse.ArgumentParser(
+        description='Check any mismatch on registers.'
+                    ' Written by Yazhou He.'
+    )
+    parser.add_argument(
+        '-preservation_storage_csv',
+        help='the correct mapping csv'
+    )
+    parser.add_argument(
+        '-register_csv',
+        help='the to-be-checked csv', required=True
+    )
+    parsed_args = parser.parse_args(args_)
+    return parsed_args
+
+def main(args_):
+    args = parse_args(args_)
+    file1 = args.preservation_storage_csv
+    file2 = args.register_csv
+
+    while not file1:
+        try:
+            ps_path = input('\nInput the path of preservation storage:\n\n')
+            file1 = get_ps_list.main(['-i', ps_path])
+        except:
+            file1 = ''
+            print('Cannot get list from input path.')
+
+    with open(file1, 'r', encoding='utf-8') as f1:
+        reader1 = csv.DictReader(f1)
+        mapping1 = defaultdict(list)
+        mapping2 = defaultdict(list)
+        for row in reader1:
+            mapping1[row['accession number']] = row['object entry number']
+            mapping2[row['object entry number']] = row['accession number']
+        mapping1 = dict(sorted(mapping1.items(), key=operator.itemgetter(0)))
+        mapping2 = dict(sorted(mapping2.items(), key=operator.itemgetter(0)))
+        
+    
+    with open(file2, 'r', encoding='utf-8') as f2:
```

Comment:
could this just be a function as it's all duplicate code?

Reply:
fixed, this part is easier than the below one.